src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/haflows/IntentionalRerouteSpec.groovy

```diff
@@ -0,0 +1,225 @@
+package org.openkilda.functionaltests.spec.flows.haflows
+
+import static org.junit.jupiter.api.Assumptions.assumeTrue
+import static org.openkilda.testing.Constants.WAIT_OFFSET
+
+import org.openkilda.functionaltests.HealthCheckSpecification
+import org.openkilda.functionaltests.extension.failfast.Tidy
+import org.openkilda.functionaltests.helpers.HaFlowHelper
+import org.openkilda.functionaltests.helpers.PathHelper
+import org.openkilda.functionaltests.helpers.Wrappers
+import org.openkilda.messaging.info.event.PathNode
+import org.openkilda.messaging.payload.flow.FlowState
+import org.openkilda.model.FlowEncapsulationType
+import org.openkilda.northbound.dto.v2.haflows.HaFlowPaths
+import org.openkilda.northbound.dto.v2.haflows.HaFlowRerouteResult
+import org.openkilda.northbound.dto.v2.haflows.HaSubFlow
+import org.openkilda.northbound.dto.v2.haflows.HaSubFlowPath
+import org.openkilda.testing.model.topology.TopologyDefinition.Switch
+
+import org.springframework.beans.factory.annotation.Autowired
+import spock.lang.Narrative
+import spock.lang.Shared
+
+@Narrative("Verify that on-demand HA-haFlow reroute operations are performed accurately.")
+class IntentionalRerouteSpec extends HealthCheckSpecification {
+    @Autowired
+    @Shared
+    HaFlowHelper haFlowHelper
+
+    @Tidy
+    def "Not able to reroute to a path with not enough bandwidth available"() {
+        given: "An HA-flow with alternate paths available"
+        def swT = topologyHelper.findSwitchTripletWithAlternativesPaths()
+        assumeTrue(swT != null, "No suiting switches found")
+        def haFlow = haFlowHelper.randomHaFlow(swT)
+        haFlow.maximumBandwidth = 10000
+        haFlowHelper.addHaFlow(haFlow)
+
+        def currentPathResponse = northboundV2.getHaFlowPaths(haFlow.haFlowId)
+        def currentPaths = currentPathResponse.subFlowPaths.collect {PathHelper.convert(it.forward)}
+
+        when: "Make the current path less preferable than alternatives"
+        def alternativePaths = (swT.pathsEp1 + swT.pathsEp2).findAll { !currentPaths.contains(it) }
+        def completeCurrentPath = currentPaths.collectMany { it}
+        alternativePaths.each { pathHelper.makePathMorePreferable(it, completeCurrentPath) }
```

Comment:
1. Why `makePathNotPreferable()` doesn't work here? 
2. Here and everywhere: several 'heavy' calls (like calls to DB, REST API, Kafka, etc.) should be done asynchronously if possible. In groovy it's done easily using 
```
withPool{
someIterable.<synchronousActionNameLikeEach>Parallel{<some closure}
}
```